### DISCUSSION TITLE
Question for developers

### DISCUSSION THREAD

**Discussion by Mr-Coquin** [luyadev/luya#2184] [Questions & Problems] [answered]
We have delivered your product. Please let me know if there is an email address.

**Comment by nadar** [comment]
Hi @Mr-Coquin, sorry what? :-) Are you on the right place? We don't delivery products nor did we order products.

**Reply by Mr-Coquin** [reply]
I have a question about: luya.io 

We have installed your system on our server. All installations went smoothly without any errors. However, we have some questions regarding the project's functionality. When attempting to save something, we encounter numerous warnings and database errors. We suspect that this issue may be due to selecting English as the default language, but we need to have multilingual support on our website, specifically for Russian and Ukrainian languages. How should we address this?

We also have a question regarding the theme or template. Currently, we are using the default option, but we would like to create our own frontend. How can we implement this?

Also, we would like to know if there are any options available for installing modules or solutions. For example, payment systems, online chat support, and similar features.

**Reply by nadar** [reply]
> When attempting to save something, we encounter numerous warnings and database errors.

Unless you post more informations about the error, we are unable to help. Multilingual websites are not a problem: https://luya.io/guide/app/language.html

> We also have a question regarding the theme or template.

You can adjust the frontend templates using the views folder layout and adding your own blocks. https://luya.io/guide/cms/blocks.html

> Also, we would like to know if there are any options available for installing modules or solutions.

There are several modules, you can create your own or maybe someone of the provided modules/packages might help. https://luya.io/packages.html

**Reply by Mr-Coquin** [reply]
We chose the LUYA CMS because having an administrative panel and efficient workflow for creating and editing pages was important to us. However, we wanted to know if it's possible to extend this solution with additional packages from your system, such as Bootstrap 4 or the News Module.


The errors occurred when I tried to save something in the database using the admin interface in Russian language. However, when I performed the same operation and filled in the fields in English, the error did not occur.

**Reply by nadar** [reply]
> we wanted to know if it's possible to extend this solution with additional packages from your system, such as Bootstrap 4 or the News Module.

Yes it is :+1: 

> The errors occurred when I tried to save something in the database using the admin interface in Russian language. However, when I performed the same operation and filled in the fields in English, the error did not occur.

Unless you provide more informations we can not help.

**Reply by Mr-Coquin** [reply]
`Unless you provide more informations we can not help.`

Alright. When we start working with the Ukrainian language and encounter SQL errors again, we will write them here.

`Yes it is `

How can we implement these changes on the existing project without breaking anything? And how can we hide the Yii panel located at the bottom right corner?

**Reply by nadar** [reply]
Please read the guide luya.io and take a look at other projects and packages from the link above.

>  And how can we hide the Yii panel located at the bottom right corner?

Its the yii debug toolbar, take a look at your config, based on your env. You might start reading the Yii framework guide first: https://www.yiiframework.com/doc/guide/2.0/en

**Reply by Mr-Coquin** [reply]
I apologize for the confusion. I understand your concerns about potentially breaking the system. To create your own template (design), such as a custom homepage for the index page (currently set as the default homepage), is there any documentation or instructions available?

**Reply by nadar** [reply]
You can use html and css in your layout https://www.yiiframework.com/doc/guide/2.0/en/structure-views#layouts and block views https://luya.io/guide/cms/blocks.html

I don't understand the question actually. Just modify the views in your views folder the way you want.

**Reply by Mr-Coquin** [reply]
I see options to select a 'theme' when creating a page and a list of 'Themes' in the page settings under 'CMS.' However, I don't understand how to add them.

**Reply by nadar** [reply]
Yes, this might be a bit confusing, there is no directory of themes you can browse and use, its not wordpress. You have to do the design by our own. If you like to create a theme to share with the world or use theme just for your company, you might read this: https://luya.io/guide/app/themes.html

**Reply by Mr-Coquin** [reply]
Great, one less dark corner! Now, I need to figure out a few other things. Perhaps you can help me understand how to implement or where to read about it. There are two forms. The first form should search for records in the database and display the retrieved information to users. The second form should retrieve data from external services via GET or REST API and provide it to website visitors. How can I properly implement this? It would be perfect if you could suggest where I can find similar examples in your documentation.

**Reply by nadar** [reply]
For both tasks you can create either a new block:

+ https://luya.io/guide/cms/blocks.html (there is even an example about a form in a block: https://luya.io/guide/lessons/blockform.html)

Or you can create a module and use the module inside the cms (using the module block, or defined a page as module)

+ https://luya.io/guide/frontend/intro.html

**Reply by Mr-Coquin** [reply]
I'm reading your documentation and YII documentation, but I can't find where to disable the debug mode. I found one place, switched it to false, but the debugger remains in place.

**Reply by nadar** [reply]
You can register/unregister modules (or bootstrap/not bootstrap) based on environments, this is the example config for luya kickstarter:

https://github.com/luyadev/luya-kickstarter/blob/master/configs/config.php#L69-L101

more infos about configs => https://luya.io/guide/installation/structure.html

**Reply by Mr-Coquin** [reply]
`[ENV_PROD](https://github.com/luyadev/luya-kickstarter/blob/master/configs/config.php#L69-L101)`

I'm here trying to turn off the debug mode from the toolbar, but I can't seem to do it. It's not detecting the changes in the file, as if it's not recognizing them.

**Reply by Mr-Coquin** [reply]
???
    $config->callback(function() {
        define('YII_DEBUG', false);
        define('YII_ENV', 'prod');
    });

**Reply by Mr-Coquin** [reply]
![image](https://github.com/luyadev/luya/assets/123034940/fcfa3f45-41ee-44a9-85bc-c284c9c2075e)
When setting the default property. An error appears on the public part of the site:
Invalid Argument
An internal server error occurred.
The above error occurred while the Web server was processing your request.

Please contact us if you think this is a server error. Thank you.

**Reply by hbugdoll** [reply]
> I'm reading your documentation and YII documentation, but I can't find where to disable the debug mode. I found one place, switched it to false, but the debugger remains in place.

> I'm here trying to turn off the debug mode from the toolbar, but I can't seem to do it. It's not detecting the changes in the file, as if it's not recognizing them.

I know it's a late answer:
- Via the constant `YII_DEBUG` you can customize the displayed error messages, see [Yii 2 Guide ](https://www.yiiframework.com/doc/guide/2.0/en/runtime-handling-errors#customizing-error-display).
- The [Yii Debugger Toolbar](https://www.yiiframework.com/extension/yiisoft/yii2-debug) is displayed, if the `debug` module of class `yii\debug\Modul` is loaded through `Config::bootstrap()`.
- Perhaps you don't switch from `Config::ENV_LOCAL` to `Config::ENV_PROD` configuration and so the debugger toolbar is loaded here: `$config->bootstrap(['debug', 'gii'])`. To disable it in local environment simply remove the module from arguments: `$config->bootstrap(['gii'])`.

**Reply by hbugdoll** [reply, marked as answer]
> When setting the default property. An error appears on the public part of the site:
Invalid Argument
An internal server error occurred.
The above error occurred while the Web server was processing your request.

@Mr-Coquin This issue was fixed last week by https://github.com/luyadev/luya-module-cms/pull/406, but not yet released.
